editor/editor_file_system.cpp

```diff
@@ -724,8 +724,9 @@ bool EditorFileSystem::_update_scan_actions() {
 					//must reimport
 					reimports.push_back(full_path);
 					Vector<String> dependencies = _get_dependencies(full_path);
-					for (const String &dependency_path : dependencies) {
-						if (import_extensions.has(dependency_path.get_extension())) {
+					for (const String &dep : dependencies) {
+						const String &dependency_path = dep.contains("::") ? dep.get_slice("::", 0) : dep;
```

Comment:
The first part here is UID, not a path.

Reply:
That was my intention, because the function it ends up in works fine with uids

Reply:
https://github.com/godotengine/godot/blob/325081cb487408dee2ba9b6d7ff8eea2086a4d7a/editor/editor_file_system.cpp#L2612-L2630

Reply:
Yeah but `import_extensions.has(dep.get_extension())` will always be false, because UID has no extension.

Reply:
`get_extension` is called on the original string, which ends with the path, so it works. Thats why the concatenated string got so far down the call stack before.

Maybe we shouldn't rely on it though and just extract the path for it anyway

Reply:
> get_extension is called on the original string

Ok I missed that 🙃
It would be clearer if `dependency_path` was used for `get_extension()` though.